src/objects/level_object.gd

```diff
@@ -0,0 +1,32 @@
+class_name LevelObject
+extends Node2D
+
+enum {
+	LEVEL_END,
+	BUTTON,
+	TOGGLE,
+	GATE
+}
+
+
+@export_range(1, 6) var minimum_weight := 1
+@export_range(1, 6) var maximum_weight := 6
+
+
+func get_object_type() -> int:
+	push_error("Function not implemented.")
+	return -1
+
+
+func get_position_offset() -> Vector2:
+	push_error("Function not implemented.")
```

Comment:
We prolly don't need this, we can let zero be the default and if something has no offset it doesn't need to implement an override